packages/env/lib/wordpress.js

```diff
@@ -91,7 +91,13 @@ async function configureWordPress( environment, config ) {
 	);
 
 	// Set wp-config.php values.
-	for ( const [ key, value ] of Object.entries( config.config ) ) {
+	for ( let [ key, value ] of Object.entries( config.config ) ) {
+		// Ensure correct port setting from config when configure WP urls.
+		if ( key === 'WP_SITEURL' || key === 'WP_HOME' ) {
+			const url = new URL( value );
```

Comment:
Hi noahtallen, thanks for the quick reply and your suggestions.

In fact I thought about similar things also and was not shure if the code change goes to the right direction.

I think the install setup, meaning '--url=localhost:${port} should stay as is. This is the WP default written to the database for home and siteurl option. And the port already comes from the config setting provided by the json files. So everything should be fine, or? By default there are no WP_HOME and WP_SITEURL is set inside wp-config.php. As far as I know it is common usage to rewrite this setting via wp-config.php if needed, and the database fallback is allways available if no WP_HOME or WP_SITEURL is set.

Rewriting the port setting if already set in WP_HOME or WP_SITEURL maybe can be problematic? I am not realy shure either. In fact these params are used to write out urls in the html pages. And it has to be the provided ports because docker maps only these ports to the containers. 
Currently these port settings (if set in the WP_HOME and WP_SITEURL) would be configured for both instances, WordPress and also testsWordPress,  which never can be correct, or? 

Alternatively the rewrite only could be done when no port is set in the URL parameters (suggesting the author did not want to configure port 80  or port 443).

Reply:
I  just thought about some placeholder in the config param. Could it be a sollution? If an config author wirtes out the placeholder in the WP_HOME or WP_SITEURL parameter, it will be replaced with the port setting, otherwise nothing happens. Something like
```
{
  "config":  {
    "WP_HOME": "http://some.domain:${port}",
  "WP_SITEURL": "http://some.domain:${port}",
  }
}
```
and in wordpress.js then
```
  value = value.replace(/\$\{port\}/, port);
```
This should be backward compatible also.

Reply:
Mmmh, now I understand better, why you asked for line 83. Third alternative could be a new config param like config.domain? Used in line 83 somehow like so (with 'localhost' as default):
```
--url=${ config.domain}:${ port }
```

Reply:
> Rewriting the port setting if already set in WP_HOME or WP_SITEURL maybe can be problematic

Yeah, exposing docker on the same port is definitely a requirement.

> I just thought about some placeholder in the config param

Neat idea! My only concern about it is that I don't know how useful it would be outside of this use case, and it doesn't seem like we really need it _that_ much anyways. 

> would be configured for both instances, WordPress and also testsWordPress, which never can be correct

True. This will be resolved by #22568. At least, a user could theoretically set it differently for each environment then.